### ISSUE TITLE
Nightly introduced bug for GGUF in comfy?

### ISSUE THREAD

**Issue by cchance27** [pytorch/pytorch#137800]
### 🐛 Describe the bug

Not sure if you'll be able to help here but trying to move the nightly to test autocast support on MPS for cogvideo, but going to nightly appears to break things, i've tested it with GGUF models in comfyui it worked fine on 2.4.1 but since upgrading to nightly it completely breaks generation...

This is on a 32gb mac, loading the Q8 GGUF flux model, that normally works fine on torch 2.4.1

```
Using split attention in VAE
INFO:root:Requested to load FluxClipModel_
Requested to load FluxClipModel_
INFO:root:Loading 1 new model
Loading 1 new model
INFO:root:loaded completely 0.0 4777.53759765625 True
loaded completely 0.0 4777.53759765625 True

ggml_sd_loader:
 GGMLQuantizationType.F32      466
 GGMLQuantizationType.Q8_0     304
 GGMLQuantizationType.F16       10
INFO:root:model weight dtype torch.bfloat16, manual cast: None
model weight dtype torch.bfloat16, manual cast: None
INFO:root:model_type FLUX
model_type FLUX
INFO:root:Requested to load Flux
Requested to load Flux
INFO:root:Loading 1 new model
Loading 1 new model
INFO:root:loaded completely 0.0 12125.320556640625 True
loaded completely 0.0 12125.320556640625 True
  0%|                                                                                             | 0/8 [00:00<?, ?it/s]
/AppleInternal/Library/BuildRoots/c7c74b64-74b4-11ef-aeda-9635a580fe0d/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:891: failed assertion `[MPSNDArray, initWithBufferImpl:offset:descriptor:isForNDArrayAlias:isUserBuffer:] Error: buffer is not large enough. Must be 120324096 bytes
'
[1]    16334 abort      PYTORCH_ENABLE_MPS_FALLBACK=1 python main.py
/opt/homebrew/anaconda3/envs/py10/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

### Versions

Versions of relevant libraries:
[pip3] came-pytorch==0.1.3
[pip3] facenet-pytorch==2.6.0
[pip3] lion-pytorch==0.2.2
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.0
[pip3] onnx==1.14.0
[pip3] onnx2torch==1.5.14
[pip3] onnxruntime==1.18.0
[pip3] onnxruntime-silicon==1.16.3
[pip3] open-clip-torch==2.24.0
[pip3] pytorch-lightning==2.3.0
[pip3] rotary-embedding-torch==0.6.5
[pip3] sensevoice-onnx==1.1.0
[pip3] torch==2.6.0.dev20241011
[pip3] torchaudio==2.5.0.dev20241011
[pip3] torchmetrics==1.3.0.post0
[pip3] torchscale==0.3.0
[pip3] torchsde==0.2.6
[pip3] torchvision==0.20.0.dev20241011

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

**Comment by malfet** [comment]
It would be nice to have a more concise reproduction..

**Comment by cchance27** [comment]
i mean i can provide the workflow i'm using and model references
[Super Basic GGUF.json](https://github.com/user-attachments/files/17346384/Super.Basic.GGUF.json)

**Comment by Vargol** [comment]
I believe I've hit the same issue in invokeAI using GGUF and flux and a 2.5.0 test version installed about 30 minutes ago (currently 12 Oct 13:21 BST)

```
AppleInternal/Library/BuildRoots/5a8a3fcc-55cb-11ef-848e-8a553ba56670/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:891: failed assertion `[MPSNDArray, initWithBufferImpl:offset:descriptor:isForNDArrayAlias:isUserBuffer:] Error: buffer is not large enough. Must be 120324096 bytes
'
```
looks like its coming out of the a mul call from a linear call, linear is one of the ops that the parameters going in have to be dequantised before the call  Here's the call stack.

```
Thread 9 Crashed:
0   libsystem_kernel.dylib        	       0x19fbe2600 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x19fc1af70 pthread_kill + 288
2   libsystem_c.dylib             	       0x19fb27908 abort + 128
3   libsystem_c.dylib             	       0x19fb26c1c __assert_rtn + 284
4   Metal                         	       0x1aab478f8 MTLReportFailure.cold.1 + 48
5   Metal                         	       0x1aab23220 MTLReportFailure + 448
6   MPSCore                       	       0x1abb17dd4 0x1ababf000 + 363988
7   libtorch_cpu.dylib            	       0x11470b768 at::native::mps::Placeholder::Placeholder(MPSGraphTensor*, at::Tensor const&, NSArray<NSNumber*>*, bool, MPSDataType, bool) + 2932
8   libtorch_cpu.dylib            	       0x114730d08 at::native::mps::binaryOpTensor(at::Tensor const&, at::Tensor const&, c10::Scalar const&, at::Tensor const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, MPSGraphTensor* (at::native::mps::BinaryOpCachedGraph*, MPSGraphTensor*, MPSGraphTensor*) block_pointer) + 1976
9   libtorch_cpu.dylib            	       0x114733ea4 at::native::structured_mul_out_mps::impl(at::Tensor const&, at::Tensor const&, at::Tensor const&) + 124
10  libtorch_cpu.dylib            	       0x111790e94 c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (at::Tensor const&, at::Tensor const&), &at::(anonymous namespace)::wrapper_MPS_mul_Tensor(at::Tensor const&, at::Tensor const&)>, at::Tensor, c10::guts::typelist::typelist<at::Tensor const&, at::Tensor const&>>, at::Tensor (at::Tensor const&, at::Tensor const&)>::call(c10::OperatorKernel*, c10::DispatchKeySet, at::Tensor const&, at::Tensor const&) + 132
11  libtorch_cpu.dylib            	       0x1105b1400 at::_ops::mul_Tensor::call(at::Tensor const&, at::Tensor const&) + 292
12  libtorch_python.dylib         	       0x103c86af8 torch::autograd::THPVariable_mul(_object*, _object*, _object*) + 396
13  libtorch_python.dylib         	       0x103c08b7c _object* torch::autograd::TypeError_to_NotImplemented_<&torch::autograd::THPVariable_mul(_object*, _object*, _object*)>(_object*, _object*, _object*) + 12
14  Python                        	       0x1013b8820 method_vectorcall_VARARGS_KEYWORDS + 144
15  Python                        	       0x1014192cc vectorcall_maybe + 108
16  Python                        	       0x101416158 slot_nb_multiply + 224
17  Python                        	       0x10138fbcc binary_op1 + 88
18  Python                        	       0x101390268 PyNumber_Multiply + 36
19  Python                        	       0x10148acec _PyEval_EvalFrameDefault + 50852
20  Python                        	       0x10148c8a8 _PyEval_Vector + 116
21  Python                        	       0x1013aedc4 method_vectorcall + 168
22  Python                        	       0x1013ad998 object_vacall + 212
23  Python                        	       0x1013adbec PyObject_CallFunctionObjArgs + 44
24  libtorch_python.dylib         	       0x10448dd28 torch::handle_torch_function_no_python_arg_parser(c10::ArrayRef<_object*>, _object*, _object*, char const*, _object*, char const*, torch::TorchFunctionName) + 2288
25  libtorch_python.dylib         	       0x103fcd670 torch::detail::(anonymous namespace)::ConcretePyInterpreterVTable::dispatch(c10::OperatorHandle const&, std::__1::vector<c10::IValue, std::__1::allocator<c10::IValue>>*) const + 644
26  libtorch_cpu.dylib            	       0x10fc53824 void c10::BoxedKernel::make_boxed_function<&(anonymous namespace)::pythonFallback(c10::OperatorHandle const&, std::__1::vector<c10::IValue, std::__1::allocator<c10::IValue>>*)>(c10::OperatorKernel*, c10::OperatorHandle const&, c10::DispatchKeySet, std::__1::vector<c10::IValue, std::__1::allocator<c10::IValue>>*) + 444
27  libtorch_cpu.dylib            	       0x110592ab8 c10::impl::BoxedKernelWrapper<at::Tensor (at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&), void>::call(c10::BoxedKernel const&, c10::OperatorHandle const&, c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&) + 76
28  libtorch_cpu.dylib            	       0x112a6d5cc c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&), &torch::autograd::VariableType::(anonymous namespace)::linear(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&)>, at::Tensor, c10::guts::typelist::typelist<c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&>>, at::Tensor (c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&)>::call(c10::OperatorKernel*, c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&) + 1200
29  libtorch_cpu.dylib            	       0x112a6de20 c10::impl::make_boxed_from_unboxed_functor<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&), &torch::autograd::VariableType::(anonymous namespace)::linear(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&)>, at::Tensor, c10::guts::typelist::typelist<c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&>>, false>::call(c10::OperatorKernel*, c10::OperatorHandle const&, c10::DispatchKeySet, std::__1::vector<c10::IValue, std::__1::allocator<c10::IValue>>*) + 112
30  libtorch_cpu.dylib            	       0x10fc53d60 void c10::BoxedKernel::make_boxed_function<&(anonymous namespace)::pythonTLSSnapshotFallback(c10::OperatorHandle const&, c10::DispatchKeySet, std::__1::vector<c10::IValue, std::__1::allocator<c10::IValue>>*)>(c10::OperatorKernel*, c10::OperatorHandle const&, c10::DispatchKeySet, std::__1::vector<c10::IValue, std::__1::allocator<c10::IValue>>*) + 176
31  libtorch_cpu.dylib            	       0x110592ab8 c10::impl::BoxedKernelWrapper<at::Tensor (at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&), void>::call(c10::BoxedKernel const&, c10::OperatorHandle const&, c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&) + 76
32  libtorch_cpu.dylib            	       0x110592380 at::_ops::linear::call(at::Tensor const&, at::Tensor const&, std::__1::optional<at::Tensor> const&) + 444
33  libtorch_python.dylib         	       0x103ef9194 torch::autograd::THPVariable_linear(_object*, _object*, _object*) + 1756
34  Python                        	       0x1013f6228 cfunction_call + 60
35  Python                        	       0x1013ab954 _PyObject_MakeTpCall + 128
36  Python                        	       0x101488638 _PyEval_EvalFrameDefault + 40944
37  Python                        	       0x10148c8a8 _PyEval_Vector + 116
38  Python                        	       0x1013aedc4 method_vectorcall + 168
39  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
40  Python                        	       0x10148c8a8 _PyEval_Vector + 116
41  Python                        	       0x1013aedc4 method_vectorcall + 168
42  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
43  Python                        	       0x10148c8a8 _PyEval_Vector + 116
44  Python                        	       0x1013abbfc _PyObject_FastCallDictTstate + 96
45  Python                        	       0x101414240 slot_tp_call + 188
46  Python                        	       0x1013ab954 _PyObject_MakeTpCall + 128
47  Python                        	       0x101488638 _PyEval_EvalFrameDefault + 40944
48  Python                        	       0x10148c8a8 _PyEval_Vector + 116
49  Python                        	       0x1013aedc4 method_vectorcall + 168
50  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
51  Python                        	       0x10148c8a8 _PyEval_Vector + 116
52  Python                        	       0x1013aedc4 method_vectorcall + 168
53  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
54  Python                        	       0x10148c8a8 _PyEval_Vector + 116
55  Python                        	       0x1013abbfc _PyObject_FastCallDictTstate + 96
56  Python                        	       0x101414240 slot_tp_call + 188
57  Python                        	       0x1013ab954 _PyObject_MakeTpCall + 128
58  Python                        	       0x101488638 _PyEval_EvalFrameDefault + 40944
59  Python                        	       0x10148c8a8 _PyEval_Vector + 116
60  Python                        	       0x1013aee50 method_vectorcall + 308
61  Python                        	       0x1013ac434 _PyVectorcall_Call + 152
62  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
63  Python                        	       0x10148c8a8 _PyEval_Vector + 116
64  Python                        	       0x1013aee50 method_vectorcall + 308
65  Python                        	       0x1013ac434 _PyVectorcall_Call + 152
66  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
67  Python                        	       0x10148c8a8 _PyEval_Vector + 116
68  Python                        	       0x1013abc6c _PyObject_FastCallDictTstate + 208
69  Python                        	       0x101414240 slot_tp_call + 188
70  Python                        	       0x1013ab954 _PyObject_MakeTpCall + 128
71  Python                        	       0x101488638 _PyEval_EvalFrameDefault + 40944
72  Python                        	       0x10148c8a8 _PyEval_Vector + 116
73  Python                        	       0x1013aee50 method_vectorcall + 308
74  Python                        	       0x1013ac434 _PyVectorcall_Call + 152
75  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
76  Python                        	       0x10148c8a8 _PyEval_Vector + 116
77  Python                        	       0x1013aee50 method_vectorcall + 308
78  Python                        	       0x1013ac434 _PyVectorcall_Call + 152
79  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
80  Python                        	       0x10148c8a8 _PyEval_Vector + 116
81  Python                        	       0x1013abc6c _PyObject_FastCallDictTstate + 208
82  Python                        	       0x101414240 slot_tp_call + 188
83  Python                        	       0x1013ab954 _PyObject_MakeTpCall + 128
84  Python                        	       0x101488638 _PyEval_EvalFrameDefault + 40944
85  Python                        	       0x10148c8a8 _PyEval_Vector + 116
86  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
87  Python                        	       0x10148c8a8 _PyEval_Vector + 116
88  Python                        	       0x1013aee50 method_vectorcall + 308
89  Python                        	       0x1013ac434 _PyVectorcall_Call + 152
90  Python                        	       0x10148a310 _PyEval_EvalFrameDefault + 48328
91  Python                        	       0x10148c8a8 _PyEval_Vector + 116
92  Python                        	       0x1013aee9c method_vectorcall + 384
93  Python                        	       0x10153da0c thread_run + 168
94  Python                        	       0x1014dde00 pythread_wrapper + 48
95  libsystem_pthread.dylib       	       0x19fc1b2e4 _pthread_start + 136
96  libsystem_pthread.dylib       	       0x19fc160fc thread_start + 8
```

**Comment by Vargol** [comment]
look like it was broke in 2.5.0.dev20240817, 2.5.0.dev20240816 doesn't give the error but the image generated is noise

**Comment by Vargol** [comment]
The is also broke in the 2.5.0 test releases.

**Comment by YAY-3M-TA3** [comment]
As of Oct 11, the nightly was working with flux and comfyui.  This version is working for me.

torch                     2.6.0.dev20241011
torchvision               0.20.0.dev20241011

**Comment by jeanjerome** [comment]
Got tis error with: 

```
pytorch                   2.6.0.dev20241013        py3.10_0    pytorch-nightly
torchaudio                2.5.0.dev20241013       py310_cpu    pytorch-nightly
torchvision               0.20.0.dev20241013       py310_cpu    pytorch-nightly
```
Error is:

```
Loading 1 new model
loaded completely 0.0 12247.570556640625 True
  0%|                                                                                                                                                                  | 0/20 [00:00<?, ?it/s]huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
	- Avoid using `tokenizers` before the fork if possible
	- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
/AppleInternal/Library/BuildRoots/5a8a3fcc-55cb-11ef-848e-8a553ba56670/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:891: failed assertion `[MPSNDArray, initWithBufferImpl:offset:descriptor:isForNDArrayAlias:isUserBuffer:] Error: buffer is not large enough. Must be 120324096 bytes
'
zsh: abort      python main.py
(comfy) jeanjerome@BigBang ComfyUI % /opt/homebrew/Caskroom/miniconda/base/envs/comfy/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
```

**Comment by Vargol** [comment]
I think I've reduced thinks down to torch only

> As of Oct 11, the nightly was working with flux and comfyui. This version is working for me. Tested with flux-hyp8-Q4_0.gguf and the other GUFF models available
> 
> torch 2.6.0.dev20241011 torchvision 0.20.0.dev20241011

What version of MacOS, I suspect this bug is in the new Mac OS 15 code, a fair bit got added between the last nightly that works and the first nightly that doesn't.

**Comment by Vargol** [comment]
I think I've created none GGUF version of the code, from the GGUF code  that fails with a similar error.
It fails on  a simple multiplication. I did try saving the tensors that go into the multiplication but doing the multiplication in isolation doesn't fail even loading the non view tensors

```py
import torch

def dequantize_blocks(
    blocks: torch.Tensor, block_size: int, type_size: int, dtype: torch.dtype
) -> torch.Tensor:
    d, x = split_block_dims(blocks, 2)
    d = d.view(torch.float16).to(dtype)
    x = x.view(torch.int8)
    print("It seems to fail here on the multiplication")
    qb = d * x
    print("This does not print")
    return qb

def split_block_dims(blocks: torch.Tensor, *args):
    n_max = blocks.shape[1]
    dims = list(args) + [n_max - sum(args)]
    return torch.split(blocks, dims, dim=1)

blocks = torch.load('blocks.pt', weights_only=True);
dqb = dequantize_blocks(blocks, 32, 34, dtype=None)

```
I've had to load a tensor from file, I tried creating a zeros tensor and a ones tensor but they do not fail, this tensor does.

[blocks.zip](https://github.com/user-attachments/files/17379720/blocks.zip)

**Comment by jeanjerome** [comment]
> What version of MacOS, I suspect this bug is in the new Mac OS 15 code, a fair bit got added between the last nightly that works and the first nightly that doesn't.

I am on Sequoia (15.0.1)

**Comment by jeanjerome** [comment]
Also got the same error with your code:
```
% python
Python 3.10.15 (main, Oct  3 2024, 02:24:49) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> 
>>> def dequantize_blocks(
...     blocks: torch.Tensor, block_size: int, type_size: int, dtype: torch.dtype
... ) -> torch.Tensor:
...     d, x = split_block_dims(blocks, 2)
...     d = d.view(torch.float16).to(dtype)
...     x = x.view(torch.int8)
...     print("It seems to fail here on the multiplication")
...     qb = d * x
...     print("This does not print")
...     return qb
... 
>>> def split_block_dims(blocks: torch.Tensor, *args):
...     n_max = blocks.shape[1]
...     dims = list(args) + [n_max - sum(args)]
...     return torch.split(blocks, dims, dim=1)
... 
>>> blocks = torch.load('blocks.pt', weights_only=True);
>>> dqb = dequantize_blocks(blocks, 32, 34, dtype=None)
It seems to fail here on the multiplication
/AppleInternal/Library/BuildRoots/5a8a3fcc-55cb-11ef-848e-8a553ba56670/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:891: failed assertion `[MPSNDArray, initWithBufferImpl:offset:descriptor:isForNDArrayAlias:isUserBuffer:] Error: buffer is not large enough. Must be 26738688 bytes
'
zsh: abort      python
```

**Comment by Vargol** [comment]
Good, hopefully the PyTorch folks will start to see if they can find the issue now they don't have to use all of ComfyUI 
 LOL.

**Comment by jeanjerome** [comment]
#https://github.com/city96/ComfyUI-GGUF/issues/107

**Comment by malfet** [comment]
> ```python
> import torch
> 
> def dequantize_blocks(
>     blocks: torch.Tensor, block_size: int, type_size: int, dtype: torch.dtype
> ) -> torch.Tensor:
>     d, x = split_block_dims(blocks, 2)
>     d = d.view(torch.float16).to(dtype)
>     x = x.view(torch.int8)
>     print("It seems to fail here on the multiplication")
>     qb = d * x
>     print("This does not print")
>     return qb
> 
> def split_block_dims(blocks: torch.Tensor, *args):
>     n_max = blocks.shape[1]
>     dims = list(args) + [n_max - sum(args)]
>     return torch.split(blocks, dims, dim=1)
> 
> blocks = torch.load('blocks.pt', weights_only=True);
> dqb = dequantize_blocks(blocks, 32, 34, dtype=None)
> ```

Thank you very much, I was able to narrow it down
```python
x = torch.rand(16, 16, device='mps', dtype=torch.float16)
y = x[:,0:2].view(torch.float32)
x * y
```
or if converted to a one line reproducer
```shell
python -c "import torch; x=torch.rand(16, 16, device='mps', dtype=torch.float16); print(x[:,0:2].view(torch.float32) * x)"
```

**Comment by malfet** [comment]
I strongly suspect that culprit is https://github.com/pytorch/pytorch/pull/128393

**Comment by DenisVieriu97** [comment]
> This is caused by #128393 as applying this fixes the crash
> 
> ```diff
> diff --git a/aten/src/ATen/native/mps/OperationUtils.mm b/aten/src/ATen/native/mps/OperationUtils.mm
> index 5d83b0699ca..6837921dd7f 100644
> --- a/aten/src/ATen/native/mps/OperationUtils.mm
> +++ b/aten/src/ATen/native/mps/OperationUtils.mm
> @@ -500,7 +500,7 @@ void printTensorNDArray(const TensorBase& t) {
>    // extract the pointer to MTLBuffer from the Tensor's storage
>    id<MTLBuffer> srcBuf = getMTLBufferStorage(src);
>  
> -  static const bool is_macOS_15_0_or_newer = is_macos_13_or_newer(MacOSVersion::MACOS_VER_15_0_PLUS);
> +  static const bool is_macOS_15_0_or_newer = false; // is_macos_13_or_newer(MacOSVersion::MACOS_VER_15_0_PLUS);
>    // Use gather kernel to solve strides for macOS < 15.0
>    // Starting with macOS 15.0, MPS supports native strides direclty in the kernels
>    if (!is_macOS_15_0_or_newer || !useMPSStridedAPI) {
> ```
> 
> @DenisVieriu97 do you know what is going on here?

I would assume this part is not handled well fp16_tensor.`view(torch.float32)`, as the slicing has been tested pretty thoroughly

**Comment by malfet** [comment]
> I would assume this part is not handled well fp16_tensor.`view(torch.float32)`, as the slicing has been tested pretty thoroughly

To be frank, I wasn't aware one can use `tensor.view` as `reinterpret_cast`, I always did it by going all the way to storage

**Comment by malfet** [comment]
@jeanjerome or @Vargol can you please confirm that the issue is fixed for you in October 21st or newer nightlies?

**Comment by jeanjerome** [comment]
@malfet Works fine with `flux1-dev-Q4_0.gguf` and `pytorch 2.6.0.dev20241020 py3.10_0 pytorch-nightly`on `Sequoia 15.0.1`. Thanks!!

**Comment by malfet** [comment]
Glad to hear that, closing and cherry-picking into 2.5.1. Curious if one observes any performance gains by moving from 2.4.1 to nightly?

**Comment by Vargol** [comment]
Ha too slow , but I can confirm it worked for me too with torch-2.6.0.dev20241021-cp311-none-macosx_11_0_arm64.whl

**Comment by Vargol** [comment]
Do not know has I don't use ComfyUI but check how you're running the VAE, as the Flux VAE doesn't work with float16, like the SDXL VAE, it needs to be bfloat16 or float32 or you get black images.  
You could try reverting torch back to 2.4.1 as see if that works.

**Comment by AzmaeenAdil** [comment]
okay, will try to see both. and how can i control the float mode, i mean
where can i change it, cause not every workflow has float related nodes
maybe?

**Comment by AzmaeenAdil** [comment]
i just realized yes, in my machine, all the flux related workflow in comfy
is giving me black result, all the previous workflow that used to just work
fine, with same settings, now suddenly starts getting the black results,
where can I change the float 32 or bflot16??
[image: image.png]

**Comment by AzmaeenAdil** [comment]
i just changed the argument from the bat file of comfy, let's see, cause, I
confirmed that, all the flux based generation is giving me black output.
simple to complex any workflow that has flux model generations. i even
changed the vae as well, but, didn't fix the issue.

**Comment by AzmaeenAdil** [comment]
thanks a lot it is fixed!! after the change of argument

 [image: image.png]

**Comment by Vargol** [comment]
Looks like GGUF has been broke again in the nightlies, I'm guessing the shift functions being moved to MPS did it, but will try to narrow it down and then raise a new issue. I'm using InvokeAI to test it but will see if diffusers has the same issue as that will be safer to replicate
